src/system/SystemObject.h

```diff
@@ -204,9 +197,76 @@ class ObjectPool
 public:
     void Reset();
 
-    T * TryCreate();
     void GetStatistics(chip::System::Stats::count_t & aNumInUse, chip::System::Stats::count_t & aHighWatermark);
 
+    template <typename... Args>
+    T * TryCreate(Args &&... args)
+    {
+        T * lReturn = nullptr;
+
+        (void) static_cast<Object *>(lReturn); /* In C++-11, this would be a static_assert that T inherits Object. */
```

Comment:
would using std::is_convertible work? We are compiling using C++14 or above I believe.

Reply:
We can even use std::is_base_of if using C++14 or above